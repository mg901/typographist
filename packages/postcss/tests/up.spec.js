const run = require('./run');
const { userConfig } = require('../../../mocks');

describe('@up', () => {
  test('with breakpoints name', async () => {
    const source = `
          .test {
            @up(desktop) {
              background-color: orange;
            }
          }`;

    expect(await run(source)).toMatchSnapshot();
  });

  test('with pixels', async () => {
    const source = `
            .test {
              @up(1000px) {
                background-color: orange;
              }
            }`;
    expect(await run(source)).toMatchSnapshot();
  });
  test('with em', async () => {
    const source = `
            .test {
              @up(40em) {
                background-color: orange;
              }
            }`;
    expect(await run(source)).toMatchSnapshot();
  });

  test('with orientation `landscape`', async () => {
    const source = `
        .test {
          @up(40em):landscape {
            background-color: orange;
          }
        }`;
    expect(await run(source)).toMatchSnapshot();
  });

  test('with orientation `portrait`', async () => {
    const source = `
          .test {
            @up(40em):portrait {
              background-color: orange;
            }
          }`;
    expect(await run(source)).toMatchSnapshot();
  });
});
