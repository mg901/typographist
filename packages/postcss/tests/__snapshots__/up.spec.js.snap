// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@up with breakpoints name 1`] = `
@media (min-width: 62em) {
  .test {
    background-color: orange;
  }
}
`;

exports[`@up with em 1`] = `
@media (min-width: 40em) {
  .test {
    background-color: orange;
  }
}
`;

exports[`@up with orientation \`landscape\` 1`] = `
@media (min-width: 40em) and (orientation: landscape) {
  .test {
    background-color: orange;
  }
}
`;

exports[`@up with orientation \`portrait\` 1`] = `
@media (min-width: 40em) and (orientation: portrait) {
  .test {
    background-color: orange;
  }
}
`;

exports[`@up with pixels 1`] = `
@media (min-width: 62.5em) {
  .test {
    background-color: orange;
  }
}
`;
